library/x509_create.c

```diff
@@ -243,6 +243,8 @@ static int parse_attribute_value_hex_der_encoded(const char *s,
         return MBEDTLS_ERR_X509_ALLOC_FAILED;
     }
     /* Beyond this point, der needs to be freed on exit. */
+    unsigned char *p = der + 1;
```

Comment:
I don't like hoisting the initialisation to this point, which is pretty random in the file.

Instead, can you hoist the declaration `unsigned char *p` to the top of the function, and leave an assignment where the initialisation was?

I think that is likely to be less fragile w.r.t. subsequent modifications to this file.

Reply:
(we might want a comment on the declaration to explain why it's had to be put to the top of the function, as otherwise I can also see a tidy-minded developer moving it back at some point in the future!)